update for 05 version (#24)
zhangli-abcd/TVR-Applicability-2: draft-zdm-tvr-applicability.md

--- abstract

> Time-Variant Routing (TVR) is a routing system that can accommodate the predicted topology changes caused by internal or

Reviewer: ```suggestion
Time-Variant Routing (TVR) is a routing system that can accommodate predicted topology changes caused by internal or
```

> The Time-Variant Routing (TVR) Working Group addresses a growing need in modern network environments where

Reviewer: ```suggestion
The Time-Variant Routing (TVR) Working Group addresses a need in network environments where
```

No need to speculate about the needs + make the text age well.

predictable variations in topology - such as the restoration, activation, or loss of network elements, are
part of normal operations. This approach is essential in dynamic networks with mobile nodes, where links may
be frequently disrupted and re-established due to mobility. It is also essential in networks with highly predictable traffic
patterns, where links may be powered down to conserve or reduce energy.

This document provides examples of implementing TVR scheduling capabilities in identified use cases. It
demonstrates the applicability of the TVR data model, methods for disseminating the TVR schedules, and the
necessary IETF ancillary technologies for network environments, such as time synchronization and policy,

> that support TVR capabilities. The examples assume YANG instance data encoding as per RFC 7951 for JSON and the TVR schedule YANG modules.

Reviewer: ```suggestion
that support TVR capabilities. The examples assume YANG instance data encoding per {{?RFC7951}} for JSON and the TVR schedule YANG modules.
```

# Conventions and Definitions

{::boilerplate bcp14-tagged}

This document uses the following terms:

> * Managing Device: A centralized entity responsible for generating

Reviewer: ```suggestion
Managing Device:
: A centralized entity responsible for generating
```

and maintaining TVR schedules.  The managing device distributes
schedules to network controllers and/or managed devices using the
TVR YANG modules.  In some deployments, the managing device may
also serve as the network controller.

> * Network Controller: An entity that receives topology schedules

Reviewer: ```suggestion
Network Controller:
: An entity that receives topology schedules
```

The managing device may need to deliver node-specific schedules to network devices by TVR Schedule Node YANG
module {{Section 5.2 of ?I-D.ietf-tvr-schedule-yang}}. This is optional and only necessary when the node attribute changes
are instructed by the controller. Meanwhile, the network devices need to report their own status data to the managing device.

The managing device needs to deliver the schedules of network topology to the network controller by the TVR Network
Topology YANG module {{Section 5.3 of ?I-D.ietf-tvr-schedule-yang}}, so that the routing application in the controller
can consider the impact of topology changes on routes when calculating routes.

The network controller should deliver the route calculation results to the network devices. The format of the routing

> results depend on the protocols deployed (The typical protocols include BGP, PCEP, etc.). Routing results MAY be delivered ahead and activated using an explicit time reference (e.g., absolute UTC), or via pre-staging plus an activation trigger.

Reviewer: ```suggestion
results depend on the protocols deployed (BGP, PCEP, etc.). Routing results MAY be delivered ahead and activated using an explicit time reference (e.g., absolute UTC), or via pre-staging plus an activation trigger.
```

nodes could apply different schedules, causing routing inconsistencies, path flapping, or packet loss.

Since the all the schedules are generated by the managing devices, it is necessary to make sure that the managing

> device and managed devices within a schedule domain {{Section 2.1.1 of ?I-D.ietf-tvr-requirements}} are synchronized

Reviewer: ```suggestion
device and managed devices within a schedule domain ({{Section 2.1.1 of ?I-D.ietf-tvr-requirements}}) are synchronized
```

# Appendix A: Code Examples
{:numbered="false"}

> All examples in this appendix are intended as RFC 7951 {{?RFC7951}} JSON instance data, following the YANG instance data encoding rules for JSON.  The duration field values are expressed in seconds unless otherwise noted.

Reviewer: ```suggestion
All examples in this appendix are intended as {{?RFC7951}} JSON instance data, following the YANG instance data encoding rules for JSON.  The duration field values are expressed in seconds unless otherwise noted.
```

~~~
{: #ex-inf10 title="An example topology for Predictable Moving Vessels" artwork-align="center"}

> This scenario is quite similar to the "Mobile Satellites" example, so the TVR node YANG module JSON example and topology YANG

Reviewer: ```suggestion
This scenario is similar to the "Mobile Satellites" example, so the TVR node YANG module JSON example and topology YANG
```
